boa/src/context.rs

```diff
@@ -476,31 +444,19 @@ impl Context {
     /// Returns if strict mode is currently active.
     #[inline]
     pub fn strict(&self) -> bool {
-        matches!(self.strict, StrictType::Global | StrictType::Function)
-    }
-
-    /// Returns the strict mode type.
-    #[inline]
-    pub(crate) fn strict_type(&self) -> StrictType {
         self.strict
     }
 
-    /// Set strict type.
-    #[inline]
-    pub(crate) fn set_strict(&mut self, strict: StrictType) {
-        self.strict = strict;
-    }
-
     /// Disable the strict mode.
     #[inline]
     pub fn set_strict_mode_off(&mut self) {
-        self.strict = StrictType::Off;
+        self.strict = false;
     }
 
     /// Enable the global strict mode.
     #[inline]
     pub fn set_strict_mode_global(&mut self) {
```

Comment:
For consistency, I would call this `set_strict_mode_on()`. Or might better to have a `set_strict_mode(bool)` function.